File: 349-IntersectionOfTwoArrays/349-IntersectionOfTwoArrays.cpp
// Last updated: 3/30/2026, 2:13:19 PM
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        set<int> s1;
        for(int num : nums1) s1.insert(num);
        vector<int> result;
        for(int num : nums2){
            if(s1.contains(num)){
                result.push_back(num);
                s1.erase(num);
            }
        }
        return result;
    }
};

File: 349-IntersectionOfTwoArrays/349-IntersectionOfTwoArrays.java
// Last updated: 3/30/2026, 2:13:12 PM
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set = new HashSet<>();
        for(int num : nums1) set.add(num);
        ArrayList<Integer> arr = new ArrayList<>();
        for(int num : nums2){
            if(set.contains(num)){
                arr.add(num);
                set.remove(num);
            }
        }
        int[] result = new int[arr.size()];
        int idx = 0;
        for(int num : arr){
            result[idx] = num;
            idx++;
        }
        return result;
    }
}